### PR TITLE
Added gradient clipping to training.

### LINKED ISSUE #79: Add gradient clipping

### PR DESCRIPTION
**Pull request by BlueCrescent**
Closes #79

### CODE REVIEW

File: config_files/config_example_hf_meditron_7B_instruction.yaml [hunk 1]
```diff
@@ -12,6 +12,8 @@ settings:
     gradient_acc_steps: 1
     local_train_micro_batch_size: 1
     sequence_length: 4096
+    gradient_clipping_mode: none
```
Comment (le1nux) [inline config_files/config_example_hf_meditron_7B_instruction.yaml:15]: can we have this nested? 
```yaml
gradient_clipping:
    mode: none
    threshold: 1.0 
```

File: src/modalities/config/config.py [hunk 2]
```diff
@@ -83,6 +83,13 @@ class WandbMode(LookupEnum):
     DISABLED = "DISABLED"
 
 
+class GradientClippingMode(LookupEnum):
```
Comment (le1nux) [inline src/modalities/config/config.py:86]: maybe a short comment about what each variant does?

File: src/modalities/utils/gradient_clipping.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+from modalities.config.config import GradientClippingMode
+from modalities.models.model import NNModel
+
+
+def build_gradient_clipper(
+    gradient_clipping_mode: GradientClippingMode, gradient_clipping_threshold: float
+) -> Callable[[NNModel], None]:
+    if gradient_clipping_mode == GradientClippingMode.P2_NORM:
+        return lambda model: (clip_grad_norm_(model.parameters(), gradient_clipping_threshold, 2), None)[-1]
```
Comment (le1nux) [inline src/modalities/utils/gradient_clipping.py:13]: what is the `None` for? Also, wouldn't the combination with `[-1]` always return `None`?

**Review by le1nux — APPROVED** [review]
left a few minor things, otherwise it looks good! :-) 
Approved it already. 👍

**Review by mali-git — APPROVED** [review]

**Pull request merged**